libs/base/Data/Vect.idr

```diff
@@ -351,10 +351,18 @@ foldrImpl f e go (x::xs) = foldrImpl f e (go . (f x)) xs
 public export
 implementation Foldable (Vect n) where
   foldr f e xs = foldrImpl f e id xs
+  foldl f z [] = z
+  foldl f z (x :: xs) = foldl f (f z x) xs
 
   null [] = True
   null _ = False
 
+  foldMap f xs = go xs neutral
+    where
+      go : forall n. Vect n a -> m -> m
+      go [] z = z
+      go (x :: xs) z = go xs (z <+> f x)
+
```

Comment:
Why not just `foldMap f = foldl (\acc, elem => acc <+> f elem) neutral` as soon as `foldl` is already tail-recursive?

Reply:
This way it's using 1 fewer higher order functions which may be faster, I'll benchmark it soon.

Reply:
Turns out it's almost exactly the same speed, so I'll use that suggestion.